pkg/agent/directive_handler.go

```diff
@@ -90,11 +90,6 @@ func (dh *BasicDirectiveHandler) handleSecretSyncCommand(
 		return
 	}
 
-	if secret == nil && cmdSchema.Action == agentapi.SecretSyncCommand_SET {
```

Comment:
Why is this ok now? I would have thought the secret would need to be required when executing the Set Command?

Reply:
Hello Lance, sorry about that, we should be ok now, we identified a last use case where we need the secret to be empty, and even if we give an empty map which is not `nil`, GRPC converts that into a `nil value`. That's why I've removed this condition.